[2025-01-05T12:59:31.661+0000] {taskinstance.py:1103} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: spark_kafka_to_postgres_dag.submit_spark_job manual__2025-01-05T12:54:25.632628+00:00 [queued]>
[2025-01-05T12:59:31.667+0000] {taskinstance.py:1103} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: spark_kafka_to_postgres_dag.submit_spark_job manual__2025-01-05T12:54:25.632628+00:00 [queued]>
[2025-01-05T12:59:31.667+0000] {taskinstance.py:1308} INFO - Starting attempt 2 of 3
[2025-01-05T12:59:31.674+0000] {taskinstance.py:1327} INFO - Executing <Task(SparkSubmitOperator): submit_spark_job> on 2025-01-05 12:54:25.632628+00:00
[2025-01-05T12:59:31.677+0000] {standard_task_runner.py:57} INFO - Started process 73 to run task
[2025-01-05T12:59:31.680+0000] {standard_task_runner.py:84} INFO - Running: ['***', 'tasks', 'run', 'spark_kafka_to_postgres_dag', 'submit_spark_job', 'manual__2025-01-05T12:54:25.632628+00:00', '--job-id', '25', '--raw', '--subdir', 'DAGS_FOLDER/teams_kafka_to_spark.py', '--cfg-path', '/tmp/tmpfx4d_gvs']
[2025-01-05T12:59:31.681+0000] {standard_task_runner.py:85} INFO - Job 25: Subtask submit_spark_job
[2025-01-05T12:59:31.707+0000] {task_command.py:410} INFO - Running <TaskInstance: spark_kafka_to_postgres_dag.submit_spark_job manual__2025-01-05T12:54:25.632628+00:00 [running]> on host 4b4d3fb23fbb
[2025-01-05T12:59:31.924+0000] {taskinstance.py:1547} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='spark_kafka_to_postgres_dag' AIRFLOW_CTX_TASK_ID='submit_spark_job' AIRFLOW_CTX_EXECUTION_DATE='2025-01-05T12:54:25.632628+00:00' AIRFLOW_CTX_TRY_NUMBER='2' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-01-05T12:54:25.632628+00:00'
[2025-01-05T12:59:31.930+0000] {base.py:73} INFO - Using connection ID 'my_spark_conn' for task execution.
[2025-01-05T12:59:31.931+0000] {spark_submit.py:339} INFO - Spark-Submit cmd: spark-submit --master spark://spark-master:7077 --conf spark.executor.extraJavaOptions=-Dlog4j.configuration=log4j.properties --packages org.apache.spark:spark-sql-kafka-0-10_2.12:3.2.0 --executor-memory 2G --driver-memory 1G --name SparkKafkaToPostgres --verbose --queue root.default /usr/local/***/include/scripts/kafka_to_postgres_teams.py
[2025-01-05T12:59:31.939+0000] {spark_submit.py:490} INFO - /usr/bin/env: ‘bash’: No such file or directory
[2025-01-05T12:59:31.946+0000] {taskinstance.py:1824} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 157, in execute
    self._hook.submit(self._application)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 422, in submit
    f"Cannot execute: {self._mask_cmd(spark_submit_cmd)}. Error code is: {returncode}."
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master spark://spark-master:7077 --conf spark.executor.extraJavaOptions=-Dlog4j.configuration=log4j.properties --packages org.apache.spark:spark-sql-kafka-0-10_2.12:3.2.0 --executor-memory 2G --driver-memory 1G --name SparkKafkaToPostgres --verbose --queue root.default /usr/local/***/include/scripts/kafka_to_postgres_teams.py. Error code is: 127.
[2025-01-05T12:59:31.948+0000] {taskinstance.py:1350} INFO - Marking task as UP_FOR_RETRY. dag_id=spark_kafka_to_postgres_dag, task_id=submit_spark_job, execution_date=20250105T125425, start_date=20250105T125931, end_date=20250105T125931
[2025-01-05T12:59:31.955+0000] {standard_task_runner.py:109} ERROR - Failed to execute job 25 for task submit_spark_job (Cannot execute: spark-submit --master spark://spark-master:7077 --conf spark.executor.extraJavaOptions=-Dlog4j.configuration=log4j.properties --packages org.apache.spark:spark-sql-kafka-0-10_2.12:3.2.0 --executor-memory 2G --driver-memory 1G --name SparkKafkaToPostgres --verbose --queue root.default /usr/local/***/include/scripts/kafka_to_postgres_teams.py. Error code is: 127.; 73)
[2025-01-05T12:59:31.990+0000] {local_task_job_runner.py:225} INFO - Task exited with return code 1
[2025-01-05T12:59:32.007+0000] {taskinstance.py:2653} INFO - 0 downstream tasks scheduled from follow-on schedule check
